apps/keycloak.py

```diff
@@ -0,0 +1,379 @@
+                    "manage-realm",
+                    "manage-users",
+                    "manage-authorization",
+                    "realm-admin",
+                ]
+            },
+        },
+        {"name": "ocfstaff", "realmRoles": ["ocfstaff"]},
+        {"name": "opstaff", "realmRoles": ["opstaff"]},
+        {"name": "sorry"},
+    ],
+    "clients": clients,
+    "clientScopes": [
+        {
+            "name": "groups",
+            "protocol": "openid-connect",
+            "attributes": {
+                "include.in.token.scope": "true",
+                "display.on.consent.screen": "true",
+            },
+            "protocolMappers": [
+                {
+                    "name": "groups",
+                    "protocol": "openid-connect",
+                    "protocolMapper": "oidc-group-membership-mapper",
+                    "consentRequired": False,
+                    "config": {
+                        "full.path": "false",
+                        "id.token.claim": "true",
+                        "access.token.claim": "true",
+                        "claim.name": "groups",
+                        "userinfo.token.claim": "true",
+                    },
+                }
+            ],
+        }
+    ],
+    "defaultDefaultClientScopes": [
+        "acr",
+        "email",
+        "profile",
+        "roles",
+        "web-origins",
+        "groups",
+    ],
+    "components": {
+        "org.keycloak.storage.UserStorageProvider": [
+            {
+                "name": "ldap",
+                "providerId": "ldap",
+                "config": {
+                    "enabled": ["true"],
+                    "vendor": ["other"],
+                    "connectionUrl": ["ldaps://ldap.ocf.berkeley.edu"],
+                    "useTruststoreSpi": ["ldapsOnly"],
+                    "connectionPooling": ["true"],
+                    "authType": ["none"],
+                    "editMode": ["READ_ONLY"],
+                    "usersDn": ["ou=People,dc=OCF,dc=Berkeley,dc=EDU"],
+                    "usernameLDAPAttribute": ["uid"],
+                    "rdnLDAPAttribute": ["uid"],
+                    "uuidLDAPAttribute": ["uid"],
+                    "userObjectClasses": ["ocfAccount,account,posixAccount"],
+                    "customUserSearchFilter": [
+                        "(!(loginShell=/opt/share/utils/bin/sorried))"
+                    ],
+                    "searchScope": ["1"],
+                    "pagination": ["true"],
+                    "importEnabled": ["true"],
+                    "batchSizeForSync": ["1000"],
+                    "fullSyncPeriod": ["604800"],
+                    "changedSyncPeriod": ["86400"],
+                    "allowKerberosAuthentication": ["true"],
+                    "kerberosRealm": ["OCF.BERKELEY.EDU"],
+                    "serverPrincipal": ["HTTP/lb.ocf.berkeley.edu@OCF.BERKELEY.EDU"],
+                    "keyTab": ["/etc/keytabs/http.keytab"],
```

Comment:
How does this get set?

Reply:
i'm working on it right now, it's not working yet